File: library/src/main/java/moe/leekcake/live2dforjvm/core/type/Vector2.java
package moe.leekcake.live2dforjvm.core.type;

import moe.leekcake.live2dforjvm.core.jni.Live2DCubismCoreJNI;

public class Vector2 extends AutoPointer {
    public Vector2() {
        this(0, 0);
    }

    public Vector2(float x, float y) {
        super(Live2DCubismCoreJNI.allocateVector2(0,0));
    }

    public float getX() {
        return Live2DCubismCoreJNI.getVector2X(pointer);
    }

    public float getY() {
        return Live2DCubismCoreJNI.getVector2Y(pointer);
    }

    public void setX(float x) {
        Live2DCubismCoreJNI.setVector2X(pointer, x);
    }

    public void setY(float y) {
        Live2DCubismCoreJNI.setVector2Y(pointer, y);
    }
}

File: library/src/main/kotlin/moe/leekcake/live2dforjvm/type/Vector2.kt
package moe.leekcake.live2dforjvm.type

import moe.leekcake.live2dforjvm.Live2DCubismCoreJNI

class Vector2(val pointer: Long) {
    companion object {
        fun allocate(x: Float = 0f, y: Float = 0f): Vector2 {
            return Vector2(Live2DCubismCoreJNI.allocateVector2(x, y))
        }
    }

    var x: Float
        get() = Live2DCubismCoreJNI.getVector2X(pointer)
        set(value) = Live2DCubismCoreJNI.setVector2X(pointer, value)

    var y: Float
        get() = Live2DCubismCoreJNI.getVector2Y(pointer)
        set(value) = Live2DCubismCoreJNI.setVector2Y(pointer, value)
}